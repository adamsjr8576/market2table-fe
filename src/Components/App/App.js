import React from 'react';
import { useSelector } from 'react-redux';
import './App.scss';
import Header from '../Header/Header';
import ZipCodeForm from '../../Containers/ZipCodeForm/ZipCodeForm';
import MarketContainer from '../../Containers/MarketContainer/MarketContainer';
import MapsContainer from '../../Containers/MapsContainer/MapsContainer';
import MarketPage from '../../Containers/MarketPage/MarketPage';
import VendorSelectContainer from '../../Containers/VendorSelectContainer/VendorSelectContainer';
import { Route, Redirect } from 'react-router-dom';

const App = () => {
  const zipCode = useSelector(state => state.zipCode);

  return (
    <main className="app-main">
      <Route path='/' render={({ location }) => {
        return <Header path={location.pathname}/>
        }}
      />
      <Route exact path='/' render={({ location }) => {
        return <ZipCodeForm path={location.pathname}/>
        }}
      />
      <Route exact path='/markets' render={({ location }) => {
        return (
          <div className='market-map-container'>
            <MarketContainer />
            <MapsContainer />
            {!zipCode && <ZipCodeForm path={location.pathname}/>}
          </div>
        )
        }}
      />
      <Route exact path='/markets/:id' render={() => {
        return (
          <div className='market-map-container'>
            <MarketPage />
            <MapsContainer />
          </div>
        )
        }}
      />
      <Route exact path='/vendor/account' render={() => {
        return (
          <div className='market-map-container'>
            <VendorSelectContainer />
          </div>
        )
        }}
      />
    </main>
  );
}

export default App;
