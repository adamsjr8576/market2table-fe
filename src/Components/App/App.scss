.app-main {
  height: 100vh;
  width: 100%;
  background-image: linear-gradient(rgba(0, 0, 0, .3), rgba(0, 0, 0, .3)), url('../../images/veggies.jpg');
  background-size: cover;
  background-position: center;
}
