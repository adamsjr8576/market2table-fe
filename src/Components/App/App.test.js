import React from 'react';
import { shallow } from 'enzyme';
import App from './App';
import { useSelector, useDispatch } from 'react-redux';

const mockState = {
  zipCode: '80401'
}

jest.mock("react-redux", () => ({
  useSelector: () => mockState,
}));

describe('App', () => {

  let wrapper;

  it('should match the snapshot', () => {
    wrapper = shallow(<App />);

    expect(wrapper).toMatchSnapshot()
  });
});
