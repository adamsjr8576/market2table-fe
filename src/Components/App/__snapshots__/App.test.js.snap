// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App should match the snapshot 1`] = `
<main
  className="app-main"
>
  <Route
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/markets"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/markets/:id"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/markets/:id/vendors"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/vendor/account"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/vendor/account/form"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/vendor/account/link/:id"
    render={[Function]}
  />
  <Route
    exact={true}
    path="/markets/:id/vendors/:id"
    render={[Function]}
  />
</main>
`;
